composer.json

```diff
@@ -30,15 +30,16 @@
         "ergebnis/composer-normalize": "^2.1",
         "guzzlehttp/psr7": "^1.0",
         "henrikbjorn/phpspec-code-coverage": "^1.0",
+        "laminas/laminas-diactoros": "^2.0",
+        "laminas/laminas-zendframework-bridge": "^1.0",
```

Comment:
this is in the require-dev section.

consumers will use this library together with the implementation of their choice. i fear that we need to do this differently. we could either in the factory check if the laminas classes exist and otherwise fallback to zend. or we could leave the existing factories alone and create new LaminasDiactoros...Factory classes.

Reply:
Well, the end user would need to do `composer require laminas/laminas-diactoros`. With that, the end user will not have to change anything in their code. The only thing that will not work is if only the legacy package zend/zend-diactoros is required, but not laminas/lamians-diactoros. Don't you think it would be fine to mention this in the docs?

Reply:
If you prefer end users not to switch the library name, we could to:
```php
use Laminas\Diactoros\Request as LaminasRequest;
use Zend\Diactoros\Request as ZendRequest;

public function createRequest(...) 
{
    // this call will trigger autoload!
    if (class_exists('Laminas\Diactoros\Request')) {
        return new LaminasRequest(...);
    }

    return new ZendRequest(...);
}
```

Reply:
the problem is, if we don't do a new major version, users might just `composer upgrade` and ignore the warning about zend being abandoned. composer will install, but they application is suddenly broken.

so i think we we should do your suggestion. does laminas always provide the BC layer of extending the old zend classes, or only if the additional bc package is installed?